### DISCUSSION TITLE
useFieldArray remove() issue

### DISCUSSION THREAD

**Discussion by ravish-goel** [react-hook-form/react-hook-form#3259] [General] [answered]
Hi Bill,

when I try to invoke remove() after calling append() on a list rendered via fields.map, I can see empty objects (specified by append method) append to the fields array. It seems like append() is executing by itself. These last items don't have the id associated with them. Also please note that the length of the fields array doesn't go down the initial length which was set based on the default data. It remains 5 in my case.

<img width="1680" alt="Screenshot 2020-10-24 at 11 41 32 PM" src="https://user-images.githubusercontent.com/27752740/97089050-b84b3d00-1652-11eb-8b91-0723de72ca58.png">
<img width="1680" alt="Screenshot 2020-10-24 at 11 39 25 PM" src="https://user-images.githubusercontent.com/27752740/97089053-be411e00-1652-11eb-8e99-05d15c4442c6.png">

**Comment by bluebill1049** [comment, marked as answer]
please provide a reproducible CSB

**Reply by ravish-goel** [reply]
Hey Bill, that's because of { shouldUnregister: false }. I set it to true and the issue got resolved. Thanks :)
